rockcraft/pebble.py

```diff
@@ -229,3 +239,34 @@ def define_pebble_layer(
             )
 
         tmp_new_layer.chmod(0o777)
+
+    @staticmethod
+    def get_part_spec(build_base: str) -> dict[str, Any]:
+        """Get the partpart that providers the pebble binary for a given build base."""
+        part_spec: dict[str, Any] = Pebble.PEBBLE_PART_SPEC
+
+        if Pebble._is_focal_or_jammy(build_base):
+            part_spec = Pebble.PEBBLE_PART_SPEC_PREVIOUS
+
+        return part_spec
+
+    @staticmethod
+    def get_entrypoint(build_base: str) -> list[str]:
+        """Get the rock's entry point for a given build base."""
+        is_legacy = Pebble._is_focal_or_jammy(build_base)
+
+        pebble_path = Pebble.PEBBLE_BINARY_PATH
+        if is_legacy:
+            # Previously pebble existed in /bin/pebble
+            pebble_path = Pebble.PEBBLE_BINARY_PATH_PREVIOUS
+
+        entrypoint = [f"/{pebble_path}", "enter"]
+
+        if is_legacy:
+            entrypoint += ["--verbose"]
+
+        return entrypoint
+
+    @staticmethod
+    def _is_focal_or_jammy(build_base: str) -> bool:
+        return build_base in ("ubuntu@20.04", "ubuntu@22.04")
```

Comment:
is the build_base already transformed to use `@` during the pydantic validation? Cause if not, and someone still uses the `:` notation, this condition won't work

Reply:
yes, we [replace ":" with "@"](https://github.com/canonical/rockcraft/blob/main/rockcraft/models/project.py#L192) during validation